parser/src/genquery2_sql.cpp

```diff
@@ -52,7 +64,7 @@ namespace irods::experimental::log
 		friend class logger<genquery2>;
 	}; // class logger_config<genquery2>
 } // namespace irods::experimental::log
-#endif // IRODS_ENABLE_430_COMPATIBILITY
+#endif // IRODS_ENABLE_4XX_COMPATIBILITY
```

Comment:
That should be `42X` instead of `4XX`.

Nice catch.

Reply:
After looking at this more, this was intentional.

I felt this was the least worst of the ideas I came up with. Alternatively, I can just remove this comment.

What do you feel is a good comment/delimiter for this?

I'm honestly fine with just removing it because it deals with more complex PP logic.

Reply:
> Should we leave a more detailed comment about this? `IRODS_ENABLE_4XX_COMPATIBILITY` does not exist, as far as I can tell.

It doesn't exist. It was meant to cover both `IRODS_ENABLE_4*_COMPATIBILITY` macros when they are used within the same if-ladder.

Thoughts?

Reply:
probably easier to just do the || thing like in other places?

Reply:
Well, this particular case isn't OR-based logic.

How about `// iRODS Compatibility Preprocessor Logic`? Does that help the reader at all?

Alternatively, this could be `// IRODS_ENABLE_*_COMPATIBILITY`.